comms/core/src/protocol/messaging/outbound.rs

```diff
@@ -115,7 +115,7 @@ impl OutboundMessaging {
                 },
                 Err(err) => {
                     metrics::error_count(&peer_node_id).inc();
-                    error!(
+                    debug!(
```

Comment:
We want to log other io errors not related to the connection closing.

Take a look at `comms/core/src/protocol/messaging/error.rs`  line 66 - I believe the error kind we're missing is WriteZero (the outbound channel could not write any bytes, presumably because the connection is closed)

Reply:
We don't want to remove all errors, just the one where it's closed

Reply:
I think I understand now. I commit a new change [here](https://github.com/tari-project/tari/pull/4578/commits/3318373d3d837d5cfc3d7cb745825368ecb36144) to address it.

Maybe that don't make sense `MessagingProtocolError`, but might we can add `ErrorKind::WouldBlock` on this list too.

Reply:
WouldBlock is handled internally by the AsyncRead implementation for the substream (Yamux). We want to keep these as narrow as possible to ensure that only "normal" disconnections are debug level logs and everything else is an error.